MSAL/src/MSALAuthenticationSchemePop+Internal.h

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import <MSAL/MSAL.h>
+#import "MSALAuthenticationSchemePop.h"
+#import "MSALAuthenticationSchemeProtocolInternal.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSALAuthenticationSchemePop () <MSALAuthenticationSchemeProtocolInternal>
```

Comment:
Just curious do you think we need `MSALAuthenticationSchemePop` available to public API? Or we can make it internal only and move back your changes from '`MSALAuthenticationSchemeProtocolInternal`' to '`MSALAuthenticationSchemeProtocol`'
I think client doesn't need to know about `MSALAuthenticationSchemePop`, they only care about the `MSALResult` object we return them. `authenticationScheme` in that object is NSString, so we don't really need '`MSALAuthenticationSchemePop`' available to public API. What do you think?

Reply:
I think we need `initWithHttpMethod:requestUrl:nonce:additionalParameters` in public API to support POP, don't we? If you think it is not needed, let's discuss it.

Reply:
Ah you're right, we need it to create the signed header. So if the client knows they want AT Pop, they will use that class to get back AT Pop. Never mind my question earlier.